pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -164,12 +164,14 @@ protected ConsumerBase(PulsarClientImpl client, String topic, ConsumerConfigurat
             this.batchReceivePolicy = BatchReceivePolicy.DEFAULT_POLICY;
         }
 
-        if (batchReceivePolicy.getTimeoutMs() > 0) {
+        initReceiverQueueSize();
+    }
+
+    protected void triggerBatchReceiveTimeoutTask() {
+        if (!hasBatchReceiveTimeout() && batchReceivePolicy.getTimeoutMs() > 0) {
```

Comment:
Both `internalPinnedExecutor` and `client.timer()` threads changes `batchReceiveTimeout`, seems not thread safe.

Reply:
The timer thread will not access the `batchReceiveTimeout`

https://github.com/apache/pulsar/pull/16160/files#diff-7a1c869c4bbefdfc72fcee6074b4dfeeec318334d48d998c1cff5e09713b0834R951-R953

But I found another place `closeConsumerTasks` will also access the `batchReceiveTimeout` but can happen on the IO thread, the user's thread, but only get the `batchReceiveTimeout`, will not change it.  And we have `volatile` for `batchReceiveTimeout`.

Reply:
My bad, `doPendingBatchReceiveTask` runs in `internalPinnedExecutor`